vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ClientItemToggleEvent.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2000-2024 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.grid;
+
+import com.vaadin.flow.component.ComponentEvent;
+
+/**
+ * Event fired when the user toggles an item on the client-side.
```

Comment:
Nit: It might not be clear what is toggled (selection, row details, expand children), so could be helpful to add that it toggles the selection state of an item. The methods for adding listeners could be improved similarly. 

We can also mention that:
- This event will always be fired after the selection change event
- What it could be used for: that it provides information on selected / deselected and whether the shift key was pressed, which could be used to implement a range selection feature

Reply:
Thanks, good point, updated.

Reply:
Looks good, thanks